task1/test_task_2.py

```diff
@@ -0,0 +1,44 @@
+from selenium.webdriver.common.by import By
+import pytest
+
+# 1. Login with all combinations of credentials (correct,incorrect,empty,illegal character, too long/short)
+
+@pytest.mark.parametrize("username,passwd,expected_result", [
+    ("standard_user", "secret_sauce", "correct"),
+    ("locked_out_user", "secret_sauce", "locked_out"),
+    ("problem_user", "secret_sauce", "problem"),
+    ("performance_glitch_user", "secret_sauce", "slow"),
+    ("wrong_username", "secret_sauce", "incorrect"),
+    ("standard_user", "wrong_password", "incorrect"),
+    ("$%\"$&$", "£%^*\"£%\"^*", "incorrect"),
+    ("", "", "incorrect"),
+    ("a"*900, "a"*900, "incorrect")
+])
+
+
+def test_login_correct(get_settings, get_driver, get_xpaths, username, passwd, expected_result):
+    driver = get_driver
+    driver.get(get_settings["base_url"])
+
+    login = driver.find_element(By.XPATH, get_xpaths["login"]["username_input"])
+    password = driver.find_element(By.XPATH, get_xpaths["login"]["password_input"])
+    login_button = driver.find_element(By.XPATH, get_xpaths["login"]["login_button"])
+
+    login.send_keys(username)
+    password.send_keys(passwd)
+    login_button.click()
```

Comment:
do you want stale elements? because that's how you get stale elements

Reply:
Fixed